src/libcharon/plugins/vici/python/README.rst

```diff
@@ -22,3 +22,16 @@ Example Usage
     >>> s.get_pools()
     OrderedDict([('p1', OrderedDict([('base', b'10.0.0.0'), ('size', b'254'),
     ('online', b'0'), ('offline', b'0')]))])
+
+Timeout support
+---------------
+
+A timeout can be configured directly on the underlying socket. Please note that
+`vici` does not support resuming partial read. Thus, to avoid corrupting
+messages, the socket temporarily switches back to blocking read after the first
+byte has been received. The initially configured timeout is restored once all
+expected bytes have been received.
```

Comment:
Documenting this so prominently might give the wrong impression that setting a timeout is something users should or even have to do (the text currently does also not explain why one would set such a timeout on the socket outside of `listen()` where it already can be configured without direct access to the socket).

Reply:
I think it's important to at least mention it, but I agree that the Readme is too prominent for an edge use-case. Would it be better to mention it in the docstring of `Session.__init__()` ?

It could be like:

```python
    def __init__(self, sock=None):
        """Establish a session with a StrongSwan daemon. By default, the Session
        will connect to StrongSwan via the `/var/run/charon.vici` Unix Domain
        Socket.

        If there is a need to connect a socket in another location or set
        specific settings on the socket like a timeout, create a connection and
        pass it to the `sock` parameter.

        :param sock: Socket connected to StrongSwan (optional)
        :type sock: socket.socket
        """
```

This would address both comments.

Reply:
Yes, I like that a lot more. I'll add some editorial suggestions.